blazor/datagrid/filtering.md

```diff
@@ -124,192 +438,1066 @@ LessThanOrEqual |Checks whether the value is less than or equal to the specified
+                    Orders.Add(new OrderData(OrderID + 9, "WELLI", 33.45, "Reims", "USA"));
+                    OrderID += 9;
+                }
+            }
+            return Orders;
+        }
+
+        public int? OrderID { get; set; }
+        public string CustomerID { get; set; }
+        public double? Freight { get; set; }
+        public string ShipCountry { get; set; }
+        public string ShipCity { get; set; }
+    }
+{% endhighlight %}
+{% endtabs %}
+
+{% previewsample "https://blazorplayground.syncfusion.com/embed/hNVJZiimmQoQZjQZ?appbar=false&editor=false&result=true&errorlist=false&theme=bootstrap5" %}
+
+## Filter grid programmatically with single and multiple values using method
+
+Programmatic filtering in the Syncfusion Blazor Grid with single and multiple values allows you to apply filters to specific columns in the grid without relying on interactions through the interface.
+
+This can be achieved by utilizing the [FilterByColumnAsync](https://help.syncfusion.com/cr/blazor/Syncfusion.Blazor.Grids.SfGrid-1.html#Syncfusion_Blazor_Grids_SfGrid_1_FilterByColumnAsync_System_String_System_String_System_Object_System_String_System_Nullable_System_Boolean__System_Nullable_System_Boolean__System_Object_System_Object_) method of the Grid.
+
+The following example demostrates, how to programmatically filter the Grid using single and multiple values for the **OrderID** and **CustomerID** columns. This is accomplished by calling the `FilterByColumnAsync` method within an external button click function.
+
+{% tabs %}
+{% highlight razor tabtitle="Index.razor" %}
+@using Syncfusion.Blazor.Grids
+@using Syncfusion.Blazor.Buttons
+
+<SfButton OnClick="onSingleValueFilter">Filter with single value</SfButton>
+<SfButton style="margin-left:5px" OnClick="onMultipleValueFilter"> Filter with multiple values</SfButton>
+
+<SfGrid DataSource="@GridData" @ref="Grid" AllowFiltering="true" Height="273px">
+    <GridFilterSettings Type="Syncfusion.Blazor.Grids.FilterType.Excel"></GridFilterSettings>
+    <GridColumns>
+        <GridColumn Field=@nameof(OrderData.OrderID) HeaderText="Order ID" TextAlign="Syncfusion.Blazor.Grids.TextAlign.Right" Width="100"></GridColumn>
+        <GridColumn Field=@nameof(OrderData.CustomerID) HeaderText="Customer ID" Width="120"></GridColumn>
+        <GridColumn Field=@nameof(OrderData.ShipCity) HeaderText="Ship City" Width="100"></GridColumn>
+        <GridColumn Field=@nameof(OrderData.ShipName) HeaderText="Ship Name" Width="100"></GridColumn>
+    </GridColumns>
+</SfGrid>
+
+@code {
+
+    public List<OrderData> GridData { get; set; }
+
+    SfGrid<OrderData> Grid;
+
+    protected override void OnInitialized()
+    {
+        GridData = OrderData.GetAllRecords();
+    }
+
+    public async Task onSingleValueFilter()
+    {
+        await Grid.FilterByColumnAsync("OrderID", "equal", 10248);
+    }
+    public async Task onMultipleValueFilter()
+    {
```

Comment:
call clearing filtering before FilterBYColumnAsync. if one button is pressed, other button filter is not applied

Reply:
change demo too

Reply:
Solved